src/validation/rules/UniqueDirectivesPerLocation.js

```diff
@@ -22,12 +27,29 @@ export function duplicateDirectiveMessage(directiveName: string): string {
 /**
  * Unique directive names per location
  *
- * A GraphQL document is only valid if all directives at a given location
+ * A GraphQL document is only valid if directives, that are not marked as `isRepeatable`, at a given location
  * are uniquely named.
  */
 export function UniqueDirectivesPerLocation(
-  context: ASTValidationContext,
+  context: ValidationContext | SDLValidationContext,
```

Comment:
This is technically a breaking change. It's probably OK as I doubt anyone is using this validation explicitly yet, but definitely need to note that in our release notes.

But I think there should be a way, today, to re-order our validation contexts to provide a Schema SDL if possible. I will probably end up forking all of these validation rules to run off an AST-only schema object, because building (and extending) the current GraphQLSchema object is too slow. I won't block this on not using the GraphQLSchema, but I'd like for us to move in that direction.

Reply:
@mjmahone It's not a breaking change since `ASTValidationContext` is not a part of public API. As we agree during `14.0.0` SDL validation is an experimental feature so I didn't expose this and other new class as public API.
ATM only `ValidationContext` is exposed so if library client use only public API he shouldn't be affected by this change in any way.